### DISCUSSION TITLE
Cannot find the manifest.json file

### DISCUSSION THREAD

**Discussion by ziwer1** [django-compressor/django-compressor#1117] [General]
By default the manifest file should be stored in the same storage as the compressed files. On my machine this should be static_root/CACHE/manifest.json
Running `python manage.py compress` populates the static_root directory with compressed js and css files. But I cannot find the manifest.json file in the CACHE directory.

This is my environment:
```
django-compressor: 4.0
django version: 3.1.14
python version: 3.7.3

```
Settings.py
```
COMPRESSORS | {'css': 'compressor.css.CssCompressor', 'js': 'compressor.js.JsCompressor'}
COMPRESS_CACHEABLE_PRECOMPILERS | ()
COMPRESS_CACHE_BACKEND | 'default'
COMPRESS_CACHE_KEY_FUNCTION | '********************'
COMPRESS_CLEAN_CSS_ARGUMENTS | ''
COMPRESS_CLEAN_CSS_BINARY | 'cleancss'
COMPRESS_CLOSURE_COMPILER_ARGUMENTS | ''
COMPRESS_CLOSURE_COMPILER_BINARY | 'java -jar compiler.jar'
COMPRESS_CSS_HASHING_METHOD | 'content'
COMPRESS_DATA_URI_MAX_SIZE | 1024
COMPRESS_DEBUG_TOGGLE | None
COMPRESS_ENABLED | True
COMPRESS_FILTERS | {'css': ['compressor.filters.css_default.CssAbsoluteFilter', 'compressor.filters.cssmin.rCSSMinFilter'],  'js': ['compressor.filters.jsmin.JSMinFilter']}
COMPRESS_JINJA2_GET_ENVIRONMENT | <function CompressorConf.JINJA2_GET_ENVIRONMENT at 0x7f900f961730>
COMPRESS_MINT_DELAY | 30
COMPRESS_MTIME_DELAY | 10
COMPRESS_OFFLINE | True
COMPRESS_OFFLINE_CONTEXT | {'STATIC_URL': '/static/'}
COMPRESS_OFFLINE_MANIFEST | 'manifest.json'
COMPRESS_OFFLINE_TIMEOUT | 31536000
COMPRESS_OUTPUT_DIR | 'CACHE'
COMPRESS_PARSER | 'compressor.parser.AutoSelectParser'
COMPRESS_PRECOMPILERS | ()
COMPRESS_REBUILD_TIMEOUT | 2592000
COMPRESS_ROOT | '/com.docker.devenvironments.code/static_root'
COMPRESS_STORAGE | 'compressor.storage.CompressorFileStorage'
COMPRESS_TEMPLATE_FILTER_CONTEXT | {'STATIC_URL': '/static/'}
COMPRESS_URL | '/static/'
COMPRESS_URL_PLACEHOLDER | '/__compressor_url_placeholder__/'
COMPRESS_VERBOSE | False
COMPRESS_YUGLIFY_BINARY | 'yuglify'
COMPRESS_YUGLIFY_CSS_ARGUMENTS | '--terminal'
COMPRESS_YUGLIFY_JS_ARGUMENTS | '--terminal'
COMPRESS_YUI_BINARY | 'java -jar yuicompressor.jar'
COMPRESS_YUI_CSS_ARGUMENTS | ''
COMPRESS_YUI_JS_ARGUMENTS | ''

INSTALLED_APPS = (
    # other apps
    "compressor",
)

STATICFILES_FINDERS = (
    'django.contrib.staticfiles.finders.FileSystemFinder',
    'django.contrib.staticfiles.finders.AppDirectoriesFinder',
    # other finders..
    'compressor.finders.CompressorFinder',
)
```
Am I doing anything wrong?

**Comment by carltongibson** [comment]
You need to run the `compress` command for this... See https://django-compressor.readthedocs.io/en/stable/usage.html#offline-compression

**Reply by ziwer1** [reply]
Hi, @carltongibson I did run the compress command. That's how I got the compressed files in the static_root directory. I had made a mistake and wrote `python manage.py` instead of `python manage.py compress` in the original post.

**Reply by carltongibson** [reply]
OK. Then you'll need to verify your COMPRESS_ROOT and so on... if it's running then the manifest file is created, normally as you say, next to the compressed files.

**Reply by ziwer1** [reply]
COMPRESS_ROOT defaults to STATIC_ROOT. And as shown in my settings.py above, my COMPRESS_ROOT is in fact my STATIC_ROOT. The compressed files are there, but the manifest.json is missing. I have run `python manage.py compress` a few more times already.

**Reply by carltongibson** [reply]
Yes. My point is that you'll need to investigate what's going wrong in your case. Generally, it works as described: for instance, if I run the command in my project the manifest file is written. It's not possible to debug your project for you over GitHub. 

If you can narrow down a specific issue we can have a look, but it's likely a configuration issue somewhere. 

Sorry I can't say more here.
